featuretools/primitives/standard/transform_primitive.py

```diff
@@ -863,3 +864,46 @@ def email_address_to_domain(emails):
                 emails_df['domain'] = emails_df['email'].str.strip().str.split('@', expand=True)[1]
             return emails_df.domain.values
         return email_address_to_domain
+
+
+class NumericLag(TransformPrimitive):
+    """Shifts an array of values by a specified number of periods.
+
+    Args:
+        periods (int): The number of periods by which to shift the input.
+            Default is 1. Periods correspond to rows.
+
+        fill_value (int, float, optional): The value to use to fill in
+            the gaps left after shifting the input. Default is None.
+
+    Examples:
+        >>> lag = NumericLag()
+        >>> lag(pd.Series(pd.date_range(start="2020-01-01", periods=5, freq='D')), [1, 2, 3, 4, 5]).tolist()
+        [nan, 1.0, 2.0, 3.0, 4.0]
+
+        You can specify the number of periods to shift the values
+
+        >>> lag_periods = NumericLag(periods=3)
+        >>> lag_periods(pd.Series(pd.date_range(start="2020-01-01", periods=5, freq='D')), [1, 2, 3, 4, 5]).tolist()
+        [nan, nan, nan, 1.0, 2.0]
+
+        You can specify the fill value to use
+
+        >>> lag_fill_value = NumericLag(fill_value=100)
+        >>> lag_fill_value(pd.Series(pd.date_range(start="2020-01-01", periods=4, freq='D')), [1, 2, 3, 4]).tolist()
+        [100, 1, 2, 3]
+    """
+    name = "numeric_lag"
+    input_types = [ColumnSchema(semantic_tags={'time_index'}), ColumnSchema(semantic_tags={'numeric'})]
+    return_type = ColumnSchema(logical_type=Double, semantic_tags={'numeric'})
```

Comment:
Not sure how important it is, but should we just leave this as `return_type = ColumnSchema(semantic_tags={'numeric'})` instead? I suspect the logical type is specified to try and avoid type inference, but this will force all columns to get converted to Double, which theoretically could cause a loss of information, especially if used with very large number inputs.

I think we should try to be specific with our return logical types when possible, but if we truly don't know that the output will always be of a certain type, maybe we should let inference happen so don't end up with an unnecessary type conversion that alters our data?

Reply:
I definitely see your point. Happy to change to just the tag.

Though if I'm reading the way woodwork types are determined in the feature matrix, we may be automatically using `Double` when the `numeric` tag is present and no logical type has been specified ( I think that this is what keeps the nullability problem from showing up here): https://github.com/alteryx/featuretools/blob/main/featuretools/computational_backends/utils.py#L318

Reply:
Yeah, think you are right, and we will end up with Double regardless. I forgot about that code in `get_ww_types_from_features`. 

I'm still thinking it would be better to leave the logical type off here, so if we ever change the logic in `get_ww_types_from_features` we won't be forcing an unnecessary conversion based on the primitive return type.